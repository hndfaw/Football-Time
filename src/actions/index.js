export const leaguesAction = leaguesData => ({
  type: 'SET_LEAGUES',
  leaguesData
})

export const setTodaysMatches = todayMatchesDate => ({
  type: 'SET_TODAYS_MATCHES',
  todayMatchesDate
})

export const setOneLeagueMatches = (leagueId, matches) => ({
  type: 'SET_ONE_LEAGUE_MATCHES',
  leagueId,
  matches
})