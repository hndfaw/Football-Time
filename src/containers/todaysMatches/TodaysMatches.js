import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './todaysMatches.css';
import MatchesContainer from '../matchesContainer/MatchesContainer';
import { fetchTodaysMatches } from '../../apiCalls';
import { setTodaysMatches } from '../../actions';



export class TodaysMatches extends Component {

  changeTodaysDate = e => {
    fetchTodaysMatches(e.target.value).then(data => this.props.handleTodaysMatches(data.api.fixtures))
  }

  filteredLeagus = () => {
    const {todaysMatches, leaguesData} = this.props;

    const filteredLeagus = [];
    todaysMatches.forEach(match => {
        leaguesData.forEach(league => {
          if (league.league_id === match.league_id && !filteredLeagus.includes(league)) {filteredLeagus.push(league) }
        })
    })

    return filteredLeagus
  }

  setNumOfLeagues = (num) => {
    this.setState({numTodaysLeagues: num})
  }

  setMatches = () => {
    const league = this.filteredLeagus().map(league => {
      return (
        <section key={league.league_id} id={league.league_id}>
        <div className="league-container">
          <div className="league-logo-container">
            <img src={league.logo} alt="league logo" className="league-logo" /> 
          </div>
          <h3 className="league-name">{league.name}</h3>
        </div>
        <MatchesContainer id={league.league_id}/>
        </section>
      )
    })
    return league
  }

  
  render() {
    const msg = this.setMatches().length !== 0 ? '' : <p className="msg">There are no matches the selected day, change the date to view another day's matches.</p>

    return (
      <main className="main-todays-matches">
        <div>
           <input type="date" className="date-picker" onChange={this.changeTodaysDate}/>
        </div>
        {msg}
        {this.setMatches()} 
      </main>
    )
  }
}

TodaysMatches.propTypes = {
  leaguesData: PropTypes.array,
  todaysMatches: PropTypes.array
}

export const mapStateToProps = state => ({
  leaguesData: state.leaguesData,
  todaysMatches: state.todaysMatches
})

export const mapDispatchToProps = dispatch => ({
  handleTodaysMatches: data => dispatch(setTodaysMatches(data)),
})

export default connect(mapStateToProps, mapDispatchToProps)(TodaysMatches)