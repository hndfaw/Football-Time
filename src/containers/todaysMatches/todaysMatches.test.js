import React from 'react';
import { TodaysMatches, mapDispatchToProps} from './TodaysMatches';
import { shallow } from 'enzyme';
import { setTodaysMatches } from '../../actions';

describe('Leagues', () => {

  // it('should match the snapshot', () => {
  //   const todaysMatches = [{
  //       awayTeamLogo: "https://www.api-football.com/public/teams/2324.png",
  //       awayTeamName: "Universidad de Concepcion",
  //       event_date: "2019-07-27",
  //       fixture_id: 81886,
  //       homeTeamLogo: "https://www.api-football.com/public/teams/2330.png",
  //       homeTeamName: "Coquimbo Unido",
  //       league_id: 303,
  //       statusShort: "FT"
  //     }]
  //   let wrapper = shallow(<Leagues />);
  //   expect(wrapper).toMatchSnapshot();
  // })

  it('should dispatch with a setTodaysMatches action when handleTodaysMatches is called', () => {
    const mockDispatch = jest.fn();
    const mockMatch = [{match: 'match'}];
    const mockAction = setTodaysMatches(mockMatch);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.handleTodaysMatches(mockMatch);
    expect(mockDispatch).toHaveBeenCalledWith(mockAction);
  })

})


