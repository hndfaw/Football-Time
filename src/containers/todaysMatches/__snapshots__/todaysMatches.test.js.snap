// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues should match the snapshot 1`] = `
<main
  className="main-todays-matches"
>
  <div>
    <input
      className="date-picker"
      onChange={[Function]}
      type="date"
    />
  </div>
  <p
    className="msg"
  >
    There are no matches the selected day, change the date to view another day's matches.
  </p>
</main>
`;
